src/ccl_f2d.c

```diff
@@ -8,6 +8,88 @@
 
 #include "ccl.h"
 
+ccl_f2d_t *ccl_f2d_t_copy(ccl_f2d_t *f2d_o, int *status)
```

Comment:
I know we don't have C tests so much, but this function seems like it needs one. Can we expose this interface in python and test it there maybe?

Reply:
Humm, how about exposing a python function that updates the non-linear power spectrum and then testing that the non-linear spectrum matches the input one? Exposing this function would be more useful in the future than an abstract function that copies python-level Pk2Ds, I think.

Reply:
Sure! As long as we test it somewhere.

Reply:
OK, but wait, we are already testing this when we check that the updated non-linear power spectrum matches the halo model implementation, right? I.e. what I just implemented here: https://github.com/LSSTDESC/CCL/blob/b9f85eb9fd7e478935edd6d3e03dadcea5618ca8/pyccl/tests/test_power.py#L40

Reply:
Did we check that all of the internals are copied correctly?

Reply:
No, good point. Let me do that now.

Reply:
Done here: https://github.com/LSSTDESC/CCL/blob/e457cfdd2892d1ea7b670d98276a0a592487b573/pyccl/tests/test_power.py#L17